### ISSUE TITLE
Offline translator 

### ISSUE THREAD

**Issue by Cheesy1576** [futo-org/android-keyboard#456]
### Use Cases

Gboard have translation option and it have it's uses, but the app needs to be in online mode for it to work and that's not gonna happen for me, so it would be cool if there was an offline translator like this in this keyboard.

**Comment by my-garbage** [comment]
You don't need to implement translation that's very space intensive.

You could like google connect it with something like OfflineTranslate https://github.com/DavidVentura/offline-translator/issues/170.

So the models and the whole Translationpart is done by a dedicated app.

The user would also benefit from Offline Translators ability to work as overlay and translate the chat. Translation itself shouldn't be a Keyboard feature because you need the translation anyway also to understand the other side. Why have to store the models 2 times.

**Comment by my-garbage** [comment]
Here would be an example how to do that https://github.com/dic1911/Momogram/tree/c677dafa07d358025886e8d4aba5e68bb668a7aa/TMessagesProj/src/main/aidl/dev/davidv/translator

**Comment by MattSolo451** [comment]
> You don't need to implement translation that's very space intensive.
> 
> You could like google connect it with something like OfflineTranslate [DavidVentura/offline-translator#170](https://github.com/DavidVentura/offline-translator/issues/170).
> 
> So the models and the whole Translationpart is done by a dedicated app.
> 
> The user would also benefit from Offline Translators ability to work as overlay and translate the chat. Translation itself shouldn't be a Keyboard feature because you need the translation anyway also to understand the other side. Why have to store the models 2 times.

I agree 100%. This feature could make the keyboard not just better, but also more competitive with Gboard and Swiftkey. Unfortunately, the developers are in no hurry.
